sigar/src/main/java/io/crate/monitor/SigarExtendedNodeInfo.java

```diff
@@ -33,15 +35,31 @@
 public class SigarExtendedNodeInfo implements ExtendedNodeInfo {
 
     private final SigarService sigarService;
+    private final NodeEnvironment nodeEnvironment;
     private final Map<File, FileSystem> fileSystems = Maps.newHashMap();
+    private final ExtendedOsStatsCache osStatsCache;
+    private final ExtendedFsStatsCache fsStatsCache;
+    private final ExtendedNetworkStatsCache networkStatsCache;
+    private final ExtendedProcessCpuStatsCache processCpuStatsCache;
+
+    private static final TimeValue PROBE_CACHE_TIME = TimeValue.timeValueMillis(500L);
 
     @Inject
-    public SigarExtendedNodeInfo(SigarService sigarService) {
+    public SigarExtendedNodeInfo(SigarService sigarService, NodeEnvironment nodeEnvironment) {
         this.sigarService = sigarService;
+        this.nodeEnvironment = nodeEnvironment;
+        this.osStatsCache = new ExtendedOsStatsCache(PROBE_CACHE_TIME, osStatsProbe());
+        this.fsStatsCache = new ExtendedFsStatsCache(PROBE_CACHE_TIME, fsStatsProbe());
+        this.networkStatsCache = new ExtendedNetworkStatsCache(PROBE_CACHE_TIME, networkStatsProbe());
+        this.processCpuStatsCache = new ExtendedProcessCpuStatsCache(PROBE_CACHE_TIME, processCpuStatsProbe());
     }
 
     @Override
-    public ExtendedNetworkStats networkStats() {
+    public synchronized ExtendedNetworkStats networkStats() {
```

Comment:
I took it from there: https://github.com/elastic/elasticsearch/blob/1.7/src/main/java/org/elasticsearch/monitor/os/OsService.java#L57